ekncontent/ekncontent/eknc-utils.c

```diff
@@ -370,6 +379,96 @@ eknc_get_data_dir (const gchar *app_id)
 
+static GList *
+databases_dirs_from_metadata (const gchar *flatpak_path, const gchar *app_id)
+{
+  GList *list = NULL;
+  g_autoptr(GKeyFile) metakey = NULL;
+  g_auto(GStrv) groups = NULL;
+  g_autofree gchar *metadata_path = NULL;
+  g_autofree gchar *runtime = NULL;
+  g_auto(GStrv) components = NULL;
+  gchar *arch = NULL;
+
+  metadata_path = g_build_filename (flatpak_path, "flatpak",
+                                    "app", app_id,
+                                    "current", "active", "metadata",
+                                    NULL);
+  metakey = g_key_file_new ();
+  if (!g_key_file_load_from_file (metakey, metadata_path, G_KEY_FILE_NONE, NULL))
+    return NULL;
+
+  runtime = g_key_file_get_string (metakey, "Application", "runtime", NULL);
+  components = g_strsplit (runtime, "/", 3);
+  arch = components[1];
+
+  groups = g_key_file_get_groups (metakey, NULL);
+  for (int i = 0; groups[i] != NULL; i++)
+    {
+      g_autoptr(GFile) extension_dir = NULL;
+      gchar *extension_name = NULL;
+      g_autofree gchar *extension_version = NULL;
+      g_autofree gchar *extension_path = NULL;
+
+      if (!g_str_has_prefix (groups[i], FLATPAK_EXTENSION_GROUP))
+        continue;
+
+      extension_name = (extension_name = (groups[i] + strlen (FLATPAK_EXTENSION_GROUP)));
+      if (!g_str_has_prefix (extension_name, app_id))
+        continue;
+
+      extension_version = g_key_file_get_string (metakey, groups[i], "version", NULL);
+      if (extension_version == NULL)
+        continue;
+
+      extension_path = g_build_filename (flatpak_path, "flatpak",
+                                         "runtime", extension_name,
+                                         arch, extension_version,
+                                         "active", "files",
+                                          NULL);
+      extension_dir = g_file_new_for_path (extension_path);
+      if (!g_file_query_exists (extension_dir, NULL))
+        continue;
+
+      list = g_list_prepend (list, g_file_dup(extension_dir));
+    }
+
+  return list;
+}
+
+/**
+ * eknc_get_extensions_dirs:
+ * @app_id: knowledge app ID, such as "com.endlessm.health-es"
+ *
+ * Searches for all the extensions directories
+ *
+ * This function searches through the system directories for all
+ * the extentesion directories that belong to this app.
+ *
+ * Return value: (element-type GFile) (transfer full): list of #GFile
+ */
+GList *
+eknc_get_extensions_dirs (const gchar *app_id)
+{
+  GList *list = NULL;
+  g_autofree gchar *system_path = NULL;
+  g_autofree gchar *split_path = NULL;
+
+  system_path = g_build_filename ("/var", "lib",
+                                  NULL);
+  list = databases_dirs_from_metadata (system_path, app_id);
+  if (list != NULL)
+    return list;
+
+  split_path = g_build_filename ("/var", "endless-extra",
+                                 NULL);
+  list = databases_dirs_from_metadata (split_path, app_id);
+  if (list != NULL)
+    return list;
```

Comment:
No need for this check, you can simply
`return databases_dirs_from_metadata ("/var/endless-extra", app_id);`

Also, dont you need to return both list concatenated ?

Reply:
These are exclusive conditions, the same logic is implemented in `eknc_get_data_dir` and I checked in my mini, the apps and runtimes are installed in `/var/endless-extra`.

If it wasn't the case then it would mean that our search-provider were historically looking at the wrong version of the content.